File: src/main/kotlin/com/stannis/dataModel/statementTypes/FieldReference.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.Arguments
import com.stannis.dataModel.NameInterface
import com.stannis.dataModel.Statement

data class FieldReference(
    override val `$type`: String? = "FieldReference",
    var fieldName: NameInterface?,
    var fieldOwner: Arguments?
) : Statement, Arguments, NameInterface

File: Interpreter/Interpreter/Models/serialize/statementTypes/FieldReference.cs
namespace Interpreter.Models.serialize.statementTypes{

    public class FieldReference : IStatement, INameInterface
    {
        public IStatement fieldName { set; get; }
        
        public IStatement fieldOwner { set; get; }

        public string GetWrittenName()
        {
            return fieldName is INameInterface ? (fieldName as INameInterface).GetWrittenName() : "";
        }
    }

};